### PR TITLE
fix: secret messages right next to each other

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

File: packages/web/src/package/observers/invisible/secret.ts [hunk 1]
```diff
@@ -56,17 +58,14 @@ export function decodeFromText(text: string) {
   const invisibleMessages = text
     .match(INVISIBLE_REGEX)
     ?.filter((m) => m.length > 8);
-  return invisibleMessages?.map(decodeMessage) || [];
+  const result = [];
+  invisibleMessages?.map(decodeMessage).forEach((message) => {
+    result.push(message);
+  });
+
+  return result;
```
Comment (Anty0) [inline packages/web/src/package/observers/invisible/secret.ts:66]: Why this change? The previous version seems to do the same thing essentially. (Am I missing something?)

File: packages/web/src/package/observers/invisible/secret.ts [hunk 2]
```diff
@@ -25,6 +25,8 @@ function padToWholeBytes(binary: string) {
 }
 
 export function encodeMessage(text: string) {
+  // insert a message end character
+  // so we can distinguish two secret messages right next to each other
```
Comment (Anty0) [inline packages/web/src/package/observers/invisible/secret.ts:29]: Rogue comment

File: packages/web/src/package/observers/invisible/InvisibleWrapper.ts [hunk 3]
```diff
@@ -49,31 +63,42 @@ export function InvisibleWrapper({ fullKeyEncode }: Props): WrapperMiddleware {
...
+          return message;
+        } else {
+          const valueCode = Number(message);
+          return keyMemory.numberToValue(valueCode);
+        }
+      });
+  }
+
+  function encodeWithSeparator(message: string) {
+    return encodeMessage(message + MESSAGE_END);
```
Comment (Anty0) [inline packages/web/src/package/observers/invisible/InvisibleWrapper.ts:82]: Maybe we can check here if the message already contains an END character and print a warning (only in some kind of debug mode). Or is this a bad idea?

**Review by Anty0 — CHANGES_REQUESTED** [review]

File: packages/web/src/package/observers/invisible/InvisibleWrapper.ts (hunk 3 above)
Comment (stepan662) [reply packages/web/src/package/observers/invisible/InvisibleWrapper.ts:82]: This is not facing the user directly, so I don't think it's necessary. The concern basically is that if someone implements a third party copy of this for e.g. PHP, he needs to know that newline is a separator, but we won't prevent that here, only by describing it well.
